ui/raidboss/data/05-shb/trial/ruby_weapon-ex.js

```diff
@@ -258,6 +296,105 @@ export default {
+      run: (data, matches) => {
+        const isBlue = matches.effectId.toUpperCase() === '8A1';
+        data.colorToImageId = data.colorToImageId || {};
+        data.colorToImageId[isBlue ? 'blue' : 'red'] = matches.targetId;
+      },
+    },
+    {
+      id: 'RubyEx Image Chariot Dynamo Collect',
+      // Lunar Dynamo = 4EB0
+      // Iron Chariot = 4EB1
+      netRegex: NetRegexes.startsUsing({ source: 'Raven\'s Image', id: ['4EB0', '4EB1'] }),
+      netRegexDe: NetRegexes.startsUsing({ source: 'Naels Trugbild', id: ['4EB0', '4EB1'] }),
+      netRegexFr: NetRegexes.startsUsing({ source: 'Spectre De Nael', id: ['4EB0', '4EB1'] }),
+      netRegexJa: NetRegexes.startsUsing({ source: 'ネールの幻影', id: ['4EB0', '4EB1'] }),
+      netRegexCn: NetRegexes.startsUsing({ source: '奈尔的幻影', id: ['4EB0', '4EB1'] }),
+      netRegexKo: NetRegexes.startsUsing({ source: '넬의 환영', id: ['4EB0', '4EB1'] }),
+      run: (data, matches) => {
+        data.imageIdToAction = data.imageIdToAction || {};
+        data.imageIdToAction[matches.sourceId] = matches.id;
+      },
+    },
+    {
+      id: 'RubyEx Image Chariot Dynamo',
+      // Lunar Dynamo = 4EB0
+      // Iron Chariot = 4EB1
+      netRegex: NetRegexes.startsUsing({ source: 'Raven\'s Image', id: ['4EB0', '4EB1'], capture: false }),
+      netRegexDe: NetRegexes.startsUsing({ source: 'Naels Trugbild', id: ['4EB0', '4EB1'], capture: false }),
+      netRegexFr: NetRegexes.startsUsing({ source: 'Spectre De Nael', id: ['4EB0', '4EB1'], capture: false }),
+      netRegexJa: NetRegexes.startsUsing({ source: 'ネールの幻影', id: ['4EB0', '4EB1'], capture: false }),
+      netRegexCn: NetRegexes.startsUsing({ source: '奈尔的幻影', id: ['4EB0', '4EB1'], capture: false }),
+      netRegexKo: NetRegexes.startsUsing({ source: '넬의 환영', id: ['4EB0', '4EB1'], capture: false }),
+      delaySeconds: 0.1,
+      suppressSeconds: 1,
+      response: (data, _matches, output) => {
+        // cactbot-builtin-response
+        output.responseOutputStrings = {
+          text: {
+            en: '${dir} (${suffix})',
+          },
+          out: Outputs.out,
+          in: Outputs.in,
+          blueSuffix: {
+            en: 'blue',
+          },
+          redSuffix: {
+            en: 'red',
+          },
+          bothSuffix: {
+            en: 'both',
+          },
+        };
+
+        if (!data.colorToImageId || !data.imageIdToAction)
+          return;
+
+        const myColor = data.colors && data.colors[data.me];
+
+        const colorToAction = {};
+        for (const color of ['blue', 'red']) {
+          const id = data.colorToImageId[color];
+          if (!id)
+            continue;
+          const action = data.imageIdToAction[id];
+          if (!action)
+            continue;
+          colorToAction[color] = action;
+        }
+
+        const numAdds = Object.keys(colorToAction).length;
+        let suffix = '';
+        let actionId = null;
```

Comment:
Can this be `let actionId = ''` or `let actionId;` just for slightly less brain at TypeScript conversion time?